File: problems/p/pyramids/pyramids.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    n = int(input())
    side = 1
    height = 0
    while (n >= (side**2)):
        n -= side ** 2
        side += 2
        height += 1
    print(height)

if __name__ == '__main__':
    main()

File: problems/p/pyramids/pyramids.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>
#include<cmath>

using namespace std;

int main() {
    int n, side, height, blocks;

    cin >> n;

    side = 1;
    height = 0;
    blocks = 1;

    while (n >= blocks) {
        n -= blocks;
        side += 2;
        blocks = pow(side, 2);
        height++;
    }

    cout << height << endl;

}